src/main/java/im/status/keycard/SecureChannel.java

```diff
@@ -176,6 +199,60 @@ public void mutuallyAuthenticate(APDU apdu) {
     respond(apdu, len, ISO7816.SW_NO_ERROR);
   }
 
+  /**
+   * Processes the LOAD_CERTS command. Copies the APDU buffer into `certs`.
+   * This function expects a DER signature and may only be called once.
+   * @param apdu the JCRE-owned APDU object.
+   */
+  public void loadCert(APDU apdu) {
+    byte[] apduBuffer = apdu.getBuffer();
+
+    if (idCertStatus != ID_CERTIFICATE_EMPTY) {
+      // Card cert may only be set once and never overwritten
+      ISOException.throwIt(ISO7816.SW_COMMAND_NOT_ALLOWED);
+    }
+
+    // Make sure the received certificate is appropriate length
+    if (apduBuffer[ISO7816.OFFSET_LC] != (byte) (CERTIFICATE_LEN)) {
+      ISOException.throwIt(ISO7816.SW_DATA_INVALID);
+    }
+
+    // Save the certificate
+    JCSystem.beginTransaction();
+    Util.arrayCopy(apduBuffer, ISO7816.OFFSET_CDATA, idCertificate, (short) 0, CERTIFICATE_LEN);
+    idCertStatus = ID_CERTIFICATE_LOCKED;
+    JCSystem.commitTransaction();
+  }
+  
+  /**
+   * Processes the IDENTIFY_CARD command. Returns the card public key, and a signature on the
+   * challenge salt, to prove ownership of the key.
+   * @param apdu the JCRE-owned APDU object.
+   */
+  public void identifyCard(APDU apdu) {
+    byte[] apduBuffer = apdu.getBuffer();
+
+    // Ensure the received challenge is appropriate length
+    if (apduBuffer[ISO7816.OFFSET_LC] != MessageDigest.LENGTH_SHA_256) {
+      ISOException.throwIt(ISO7816.SW_DATA_INVALID);
+    }
+
+    // Copy card ID pubKey to the response buffer
+    short responseOffset = (short) ISO7816.OFFSET_CDATA + (short) MessageDigest.LENGTH_SHA_256;
+    short off = responseOffset;
+    ECPublicKey pk = (ECPublicKey) idKeypair.getPublic();
+    short len = pk.getW(apduBuffer, off);
+    off += len;
+
+    // Sign the challenge and copy signature to response buffer
+    eccSig.init(idKeypair.getPrivate(), Signature.MODE_SIGN);
+    len = eccSig.signPreComputedHash(apduBuffer, (short) ISO7816.OFFSET_CDATA, MessageDigest.LENGTH_SHA_256, apduBuffer, off);
```

Comment:
I think you need to do crypto.fixS on your signature output, as is done here:

https://github.com/status-im/status-keycard/blob/master/src/main/java/im/status/keycard/KeycardApplet.java#L1285

Reply:
It is not needed. See above